### PR TITLE
bugfix Idefics3 processor - handle gracefully cases with text and no images

### PR DESCRIPTION
**Pull request by mfarre**
# What does this PR do?

Fixing Idefics3 processor to work with batches that do not include images

## Who can review?
@andimarafioti

### CODE REVIEW

**Comment by andimarafioti** [comment]
LGTM!

**Review by andimarafioti — APPROVED** [review]
LGTM!

**Comment by andimarafioti** [comment]
Let's wait for @yonigozlan to be sure!

File: src/transformers/models/idefics3/processing_idefics3.py
Comment (yonigozlan) [inline src/transformers/models/idefics3/processing_idefics3.py:328]:
```suggestion
        elif text is not None:
            if any(n_images_in_text):
                raise ValueError(
                    f"Found {sum(n_images_in_text)} {self.image_token.content} tokens in the text but no images were passed."
                )
            text_inputs = self.tokenizer(text=text, **output_kwargs["text_kwargs"])
```

I'd add this otherwise we won't have an error if there are image tokens in the conversation, but no images passed

**Review by yonigozlan — COMMENTED** [review]
Yes that looks much better! Just suggested a check to avoid hallucinations like this:
```python
messages = [
    {
        "role": "user",
        "content": [
            {"type": "image"},
            {"type": "text", "text": "What do we see in this image?"},
        ]
    }
]
prompt = processor.apply_chat_template(messages, add_generation_prompt=True)
inputs = processor(text=prompt, return_tensors="pt")
inputs = {k: v.to(DEVICE) for k, v in inputs.items()}


# Generate
generated_ids = model.generate(**inputs, max_new_tokens=50)
generated_texts = processor.batch_decode(generated_ids, skip_special_tokens=True)

print(generated_texts)

['User:What do we see in this image?\nAssistant: The image depicts a scene from a historical or fictional setting, likely from the medieval period, given the attire and architecture. The central focus is on a large, ornate gate, which appears to be the entrance to a castle or a fortified structure.']
```

also in `Idefics3ProcessorTest` could we add a test for text only inference and one to check that an error is raised if we have an image in the conversation, but no images are passed to the processor?

**Comment by mfarre** [comment]
thanks @andimarafioti 
thanks @yonigozlan if you give me your LGTM I will merge the changes: I followed your suggestions adding some tests and adding your code proposal.

> Yes that looks much better! Just suggested a check to avoid hallucinations like this:
> 
> ```python
> messages = [
>     {
>         "role": "user",
>         "content": [
>             {"type": "image"},
>             {"type": "text", "text": "What do we see in this image?"},
>         ]
>     }
> ]
> prompt = processor.apply_chat_template(messages, add_generation_prompt=True)
> inputs = processor(text=prompt, return_tensors="pt")
> inputs = {k: v.to(DEVICE) for k, v in inputs.items()}
> 
> 
> # Generate
> generated_ids = model.generate(**inputs, max_new_tokens=50)
> generated_texts = processor.batch_decode(generated_ids, skip_special_tokens=True)
> 
> print(generated_texts)
> 
> ['User:What do we see in this image?\nAssistant: The image depicts a scene from a historical or fictional setting, likely from the medieval period, given the attire and architecture. The central focus is on a large, ornate gate, which appears to be the entrance to a castle or a fortified structure.']
> ```
> 
> also in `Idefics3ProcessorTest` could we add a test for text only inference and one to check that an error is raised if we have an image in the conversation, but no images are passed to the processor?

**Comment by yonigozlan** [comment]
Looks great thanks @mfarre for fixing this! LGTM for me, but let's maybe wait for @ArthurZucker 's review before merging :)

**Review by ArthurZucker — APPROVED** [review]
Thanks for fixing 🤗

**Pull request merged**
